yt/data_objects/construction_data_containers.py

```diff
@@ -1270,16 +1278,38 @@ def _calculate_flux_in_grid(self, grid, mask,
         ret.convert_to_units(self.ds.unit_system[ret_units.dimensions])
         return ret
 
+    _vertices = None
     @property
-    def triangles(self):
-        if self.vertices is None:
+    def vertices(self):
+        if self._vertices is None:
             self.get_data()
+        return self._vertices
+
+    @property
+    def triangles(self):
         vv = np.empty((self.vertices.shape[1]//3, 3, 3), dtype="float64")
+        vv = self.ds.arr(vv, self.vertices.units)
         for i in range(3):
             for j in range(3):
                 vv[:,i,j] = self.vertices[j,i::3]
         return vv
 
+    _surface_area = None
+    @property
+    def surface_area(self):
```

Comment:
Any chance we could collapse the operations here to some more in-place?  I think it's likely this would create a lot of temporaries.